### PR TITLE
📄 Add more clarity on merging of config files 

### PR DESCRIPTION
**Pull request by sid-r-singh**
## 🤔 What does this PR do ?
It adds more clarity on merging of config files when different config files for different environments are present in a Hugo site.

## 📄 Which files does this PR modify ?
File `configuration.md` located at `content/en/getting-started/configuration.md`

#### Before this PR
image

#### After this PR
image

### CODE REVIEW

File: content/en/getting-started/configuration.md [hunk 1]
```diff
@@ -77,6 +77,27 @@ foo = "bar"
...
 
 Considering the structure above, when running `hugo --environment staging`, Hugo will use every settings from `config/_default` and merge `staging`'s on top of those.
+
+Let's take an example to understand this better. Let's say you are using Google analytics for your website. This requires you to specify `googleAnalytics = "UA-XXXXXXXX"` in `config.toml`. Now consider the following scenario:
+- You don't want the Analytics code to be loaded in development i.e. in your `localhost`
+- You want to use separate googleAnalytics IDs for your production & staging environments (say):
+  - `UA-PPPPPPPP` for production
+  - `UA-SSSSSSSS` for staging
+
+This is how you need to configure tour `config.toml` files considering the above scenario:
```
Comment (sid-r-singh) [inline content/en/getting-started/configuration.md:87]: Sorry for the error, "tour" should be "your".

File: content/en/getting-started/configuration.md [hunk 2]
```diff
@@ -77,6 +77,27 @@ foo = "bar"
 ```
 
 Considering the structure above, when running `hugo --environment staging`, Hugo will use every settings from `config/_default` and merge `staging`'s on top of those.
+
+Let's take an example to understand this better. Let's say you are using Google analytics for your website. This requires you to specify `googleAnalytics = "UA-XXXXXXXX"` in `config.toml`. Now consider the following scenario:
```
Comment (coliff) [inline content/en/getting-started/configuration.md:81]: **Google analytics** should be **Google Analytics**

File: content/en/getting-started/configuration.md [hunk 3]
```diff
@@ -77,6 +77,27 @@ foo = "bar"
...
+  - `UA-PPPPPPPP` for production
+  - `UA-SSSSSSSS` for staging
+
+This is how you need to configure your `config.toml` files considering the above scenario:
+1. In `_default/config.toml` you don't need to mention `googleAnalytics` parameter at all. This ensures that no Google Analytics code is loaded in your development server i.e. when you run `hugo serve`. This works since, by default Hugo sets `Environment=development` when you run `hugo serve` which uses the config files from `_default` folder
+2. In `production/config.toml` you just need to have one line:
+
+    ```googleAnalytics = "UA-PPPPPPPP"```
+
+    You don't need to mention all other parameters like `title`, `baseURL`, `theme` etc. againg in this config file. You need to mention only those parameters which are different or new for the production environment. This is due to the fact that Hugo is going to __merge__ this on top of `_default/config.toml`. Now when you run `hugo` (build command), by default hugo sets `Environment=production`, so the `UA-PPPPPPPP` analytics code will be there in your production website
```
Comment (coliff) [inline content/en/getting-started/configuration.md:93]: typo 'againg'

Comment (sid-r-singh) [reply content/en/getting-started/configuration.md:93]: Sorry for that.

Comment (sid-r-singh) [reply content/en/getting-started/configuration.md]: It's corrected now.

File: content/en/getting-started/configuration.md (hunk 1 above)
Comment (sid-r-singh) [reply content/en/getting-started/configuration.md]: Fixed this.

File: content/en/getting-started/configuration.md (hunk 2 above)
Comment (sid-r-singh) [reply content/en/getting-started/configuration.md]: Fixed this

File: content/en/getting-started/configuration.md (hunk 3 above)
Comment (sid-r-singh) [reply content/en/getting-started/configuration.md]: Updated the screenshot as well.

**Comment by bep** [comment]
@coliff is this good to go?

**Comment by coliff** [comment]
I think the wording all looks good but now that Universal Analytics is end-of-life we should replace all the `UA-` with `G-` so they apply to Google Analytics 4.

**Review by coliff — APPROVED** [review]

**Comment by coliff** [comment]
thanks @sid-r-singh - sorry for delay in getting this merged. 👍

**Comment by sid-r-singh** [comment]
Thanks @bep @coliff for finally merging this. I am very happy that my contribution was accepted.

**Pull request merged**